pkg/resourcemanager/appinsights/api_keys_reconcile.go

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package appinsights
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/Azure/azure-service-operator/api/v1alpha1"
+	azurev1alpha1 "github.com/Azure/azure-service-operator/api/v1alpha1"
+	"github.com/Azure/azure-service-operator/pkg/errhelp"
+	"github.com/Azure/azure-service-operator/pkg/helpers"
+	"github.com/Azure/azure-service-operator/pkg/resourcemanager"
+	"github.com/Azure/azure-service-operator/pkg/secrets"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func (c *InsightsAPIKeysClient) Ensure(ctx context.Context, obj runtime.Object, opts ...resourcemanager.ConfigOption) (bool, error) {
+
+	instance, err := c.convert(obj)
+	if err != nil {
+		return false, err
+	}
+
+	options := &resourcemanager.Options{}
+	for _, opt := range opts {
+		opt(options)
+	}
+
+	if options.SecretClient != nil {
+		c.SecretClient = options.SecretClient
+	}
+
+	instance.Status.Provisioning = true
+
+	// we have reconciled this previously, check if it already exists
+	if instance.Status.ResourceId != "" {
+		idParts := strings.Split(instance.Status.ResourceId, "/")
+
+		_, err := c.GetKey(
+			ctx,
+			instance.Spec.ResourceGroup,
+			instance.Spec.AppInsights,
+			idParts[len(idParts)-1],
+		)
+		if err == nil {
+			return true, nil
+		}
+
+		return false, nil
+
+	}
```

Comment:
Remove extra blank lines #trivial-nit

Reply:
done...I think there may be a better go fmt lib that can automate this...